### PR TITLE
Add CI to verify hydration works.

### PR DESCRIPTION
**Pull request by jlewi**
* Write a golang test to verify we can hydrate the manifests

* Trigger go tests under prow

* The name of the stack was changed from GCP to generic so we need to update it.

* Related to kubeflow/gcp-blueprints#42

### CODE REVIEW

**Comment by jlewi** [comment]
This isn't ready for review yet; I need to verify that the tests actually run successfully.

**Comment by jlewi** [comment]
Logs show

```
INFO|2020-06-15T18:31:07|/src/kubeflow/testing/py/kubeflow/testing/tekton_client.py|58| Parsing ApiException body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pipelineruns.tekton.dev \"kubeflow-gcp-blueprints-presubmit-go-unittests-50-b1d4fcd-3488-c09d\" not found","reason":"NotFound","details":{"name":"kubeflow-gcp-blueprints-presubmit-go-unittests-50-b1d4fcd-3488-c09d","group":"tekton.dev","kind":"pipelineruns"},"code":404}
```

I suspect the problem is that the name is too long so it gets rejected.

Name is 67 characters long

**Comment by jlewi** [comment]
kubeflow/testing#683 was merged; lets see if that fixes the problem.

/test all

**Comment by jlewi** [comment]
go tests actually failed even though the presubmit is reported as success; probably because we aren't probably taking the junit files into account in the github status.

**Comment by Bobgy** [comment]
Thanks! This helps development a lot.
/hold
Please remove hold when you feel ready to review (to make the signal clearer)

**Comment by jlewi** [comment]
The test failure 

```
go test ./...
# _/git_gcp-blueprints/tests
package _/git_gcp-blueprints/tests (test)
	imports github.com/termie/go-shutil: cannot find package "github.com/termie/go-shutil" in any of:
	/usr/local/go/src/github.com/termie/go-shutil (from $GOROOT)
	/root/go/src/github.com/termie/go-shutil (from $GOPATH)
FAIL	_/git_gcp-blueprints/tests [setup failed]
FAIL
```

Is because we are issuing "go test ./..." in the root of the repository which doesn't have a "go.mod" file.

**Comment by jlewi** [comment]
/hold cancel

@Bobgy Tests are passing; this is ready for review.

File: kubeflow/instance/kustomize/kubeflow-apps/kustomization.yaml [hunk 1]
```diff
@@ -5,5 +5,5 @@ patchesStrategicMerge:
 - default-install-config.yaml
 - pipeline-minio-install-config.yaml
 resources:
-- ../../../upstream/manifests/stacks/gcp # {"type":"string","x-kustomize":{"setBy":"kpt","partialSetters":[{"name":"kustomize_manifests_path","value":"../../../upstream/manifests"}]}}
```
Comment (Bobgy) [inline kubeflow/instance/kustomize/kubeflow-apps/kustomization.yaml:8]: not very related to this PR, but shall we add gcp stacks back to manifests, I'm making some gcp specific components and I think a gcp stacks is needed, we cannot compose all of them in instance/ folder

**Pull request merged**